### PR TITLE
Continue implementation of Cap Cons

### PR DESCRIPTION
**Pull request by turbolent**
Work towards #2091

## Description

- Test existing implementation
- Add support for ID capability to CCF codec
- Fix borrowing ID capability

______

- [ ] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: runtime/capabilitycontrollers_test.go [hunk 1]
```diff
@@ -208,200 +215,436 @@ func TestRuntimeCapabilityControllers(t *testing.T) {
...
+                                      let resourceID = 42
+
+                                      // Arrange
+                                      Test.createAndSaveR(id: resourceID, storagePath: storagePath)
+                                      signer.link<&Test.R>(publicPath, target: storagePath)!
+
+                                      // Act
+                                      let gotCap1: Capability<&Test.R>? =
+                                          %[1]s.getCapability<&Test.R>(publicPath)
+                                      let gotCap2: Capability<&Test.R>? =
```
Comment (SupunS) [inline runtime/capabilitycontrollers_test.go:242]: Is there a test that does the opposite of this? i.e: publishing a capability through the new API, and trying to get it from the old `getCapability()` API

**Review by SupunS — APPROVED** [review]
Nice!

File: encoding/ccf/decode.go [hunk 2]
```diff
@@ -1200,18 +1214,41 @@ func (d *Decoder) decodeCapability(typ *cadence.CapabilityType, types *cadenceTy
...
+			return nil, err
+		}
+
+		return cadence.NewMeteredIDCapability(
+			d.gauge,
+			id.(cadence.UInt64),
+			address.(cadence.Address),
+			typ.BorrowType,
+		), nil
+	} else {
```
[the remark this answers is not in the document]
Comment (turbolent) [reply encoding/ccf/decode.go]: Simplified in 79d0d33b4a3ea13c1614fb6e687a2bbc4e74f93a

File: runtime/capabilitycontrollers_test.go (hunk 1 above)
Comment (turbolent) [reply runtime/capabilitycontrollers_test.go:242]: I'm not quite sure in which PR such a test is added, but the branch of the final PR has such a test case (if I'm understanding correctly): https://github.com/onflow/cadence/blob/bastian/cap-cons-11/runtime/capabilitycontrollers_test.go#L266-L344

(`getCapability` is basically just a constructor for `Capability`, it does not interact with storage at all, so calling it before or after `issue` is the same).

**Pull request merged**